frontend/test/views/AccountView.spec.js

```diff
@@ -75,4 +77,55 @@ describe('AccountView.vue', () => {
       expect(wrapper.vm.$router.push.called).to.be.true;
     });
   });
+
+  it('should toggle the secret value on click', async () => {
+    // Create a stub for the getUser method and return a mock user
+    const mockUser = {
+      clientSecret: 'testsecret',
+    };
+
+    const getUserStub = sandbox.stub(authStore, 'getUser');
+    getUserStub.returns(mockUser);
+
+    const wrapper = getMountedComponent();
+
+    expect(wrapper.vm.showSecretValue).to.be.false;
+    expect(wrapper.vm.secretValue).to.deep.equal(['<click to show>']);
+
+    // Find the section box component for the "Client Secret" section
+    const sectionBox = wrapper.findComponent({ref: 'credentials'});
```

Comment:
We rely on utilizing 'data-test' instead of ref in this code base to query component using its dom attribute.